Core/WumpusCore/GameLocations/GameLocations.cs

```diff
@@ -25,22 +25,34 @@ public enum RoomTypes
         /// <summary>
         /// The random object from the controller class.
         /// </summary>
-        Random random;
+        private Random random;
 
         public GameLocations()
         {
             rooms = new List<Room>();
             random = Controller.Controller.Random;
```

Comment:
Holding a reference here is maybe okay, but it obfuscates the true source of random and I think it would be better to grab from Controller each time. Saves memory too. <|:)